std/getopt.d

```diff
@@ -1102,6 +1108,23 @@ unittest
     getopt(args, "fb1", &fb1, "fb2", &fb2, "tb1", &tb1);
     assert(fb1 && fb2 && !tb1);
 
+    // test keepEndOfOptions
+
+    args = (["program.name", "--foo", "nonoption", "--bar", "--", "--baz"]).dup;
```

Comment:
I'm guessing the code is really old and there were circumstances calling for this awkwardness. In this PR it's probably nice to keep it consistent with what is there; it should rather be fixed wholesale in a different PR.